src/selection/mouseselectionhelper.ts

```diff
@@ -0,0 +1,86 @@
+import * as d3 from 'd3';
+import {toSelectOperation} from 'phovea_core/src/idtype';
+import {SelectOperation} from 'phovea_core/src/idtype/IIDType';
+import {fire} from 'phovea_core/src/event';
+import List from '../list';
+
+export class MouseSelectionHelper {
+  topBottom : number[];
+  mouseLeft : boolean = false;
+  constructor(private mouseListener : d3.Selection<any>, private mouseLeaveObject : d3.Selection<any>, private data: any) {
+    this.topBottom = [-1, -1];
+  }
+
+  installListeners(onClickAdd, onClickRemove) {
+    this.mouseListener.on('mousedown', (d, i) => {
+      this.resetTopbottom();
+      this.updateTopBottom(i, this.topBottom[1], this.topBottom);
+      if (toSelectOperation(<MouseEvent>d3.event) === SelectOperation.SET) {
+        this.data.clear();
+        fire(List.EVENT_BRUSH_CLEAR, this.data);
+      }
+    })
+    .on('mouseenter', (d, i) => {
+      if(this.topBottom[0] !== -1) {
+        this.removeOldSelectedElements(this.topBottom, i, onClickRemove);
+        this.updateTopBottom(this.topBottom[0], i, this.topBottom);
+        this.selectTopBottom(this.topBottom, onClickAdd);
+      }
+    })
+    .on('mouseup', (d, i) => {
+      if(this.topBottom[0] !== -1) {
+        this.updateTopBottom(this.topBottom[0], i, this.topBottom);
+        this.selectTopBottom(this.topBottom, onClickAdd);
+        this.topBottom.sort((a, b) => a - b);
+        fire(List.EVENT_BRUSHING, this.topBottom, this.data);
+        this.resetTopbottom();
+      }
+    });
+
+    this.mouseLeaveObject.on('mouseleave', (d, i) => {
+       if(this.topBottom[0] !== -1) {
+        this.selectTopBottom(this.topBottom, onClickAdd);
+        this.topBottom.sort((a, b) => a - b);
+        fire(List.EVENT_BRUSHING, this.topBottom, this.data);
+        this.resetTopbottom();
+      }
+    });
+
+    document.documentElement.addEventListener('mouseup', function(e){
+      console.log(e);
```

Comment:
Please remove the `console.log` or in this case the whole listener, since it has no function.